pipelines/data_pipelines.go

```diff
@@ -0,0 +1,38 @@
+package pipelines
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/databricks/databricks-sdk-go"
+	"github.com/databricks/databricks-sdk-go/service/pipelines"
+	"github.com/databricks/terraform-provider-databricks/common"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+)
+
+func DataSourcePipelines() *schema.Resource {
+	type pipelineData struct {
+		Name string            `json:"pipeline_name,omitempty"`
+		Ids  map[string]string `json:"ids,omitempty" tf:"computed"`
```

Comment:
```suggestion
		Ids  []string `json:"ids,omitempty" tf:"computed,slice_set"`
```

we don't have a pattern of "id-to-name" anywhere. please use the set and sort it, before returning - it'll make unit tests predictable.

1. name to id: https://registry.terraform.io/providers/databricks/databricks/latest/docs/data-sources/mws_credentials#ids
2. set: https://registry.terraform.io/providers/databricks/databricks/latest/docs/data-sources/clusters#ids
3. name to id: https://registry.terraform.io/providers/databricks/databricks/latest/docs/data-sources/jobs#ids
4. set https://registry.terraform.io/providers/databricks/databricks/latest/docs/data-sources/sql_warehouses#ids
5. name to id https://registry.terraform.io/providers/databricks/databricks/latest/docs/data-sources/mws_workspaces#ids
6. set https://registry.terraform.io/providers/databricks/databricks/latest/docs/data-sources/service_principals#application_ids
7. set https://registry.terraform.io/providers/databricks/databricks/latest/docs/data-sources/dbfs_file_paths#path_list
8. set https://registry.terraform.io/providers/databricks/databricks/latest/docs/data-sources/catalogs#ids
9. set https://registry.terraform.io/providers/databricks/databricks/latest/docs/data-sources/schemas#ids
10. set https://registry.terraform.io/providers/databricks/databricks/latest/docs/data-sources/shares#shares
11. set https://registry.terraform.io/providers/databricks/databricks/latest/docs/data-sources/tables#ids
12. set https://registry.terraform.io/providers/databricks/databricks/latest/docs/data-sources/views#ids
13. set https://registry.terraform.io/providers/databricks/databricks/latest/docs/data-sources/notebook_paths#notebook_path_list

Reply:
Refactored to return a list of ids as discussed.